substrate/primitives/src/networks.rs

```diff
@@ -10,38 +10,98 @@ use borsh::{BorshSerialize, BorshDeserialize};
 use serde::{Serialize, Deserialize};
 
 use sp_core::{ConstU32, bounded::BoundedVec};
+use sp_std::{vec, vec::Vec};
 
 #[cfg(feature = "borsh")]
 use crate::{borsh_serialize_bounded_vec, borsh_deserialize_bounded_vec};
 
-/// The type used to identify networks.
+/// The type used to identify external networks.
 #[derive(
   Clone, Copy, PartialEq, Eq, Hash, Debug, Encode, Decode, PartialOrd, Ord, MaxEncodedLen, TypeInfo,
 )]
 #[cfg_attr(feature = "std", derive(Zeroize))]
 #[cfg_attr(feature = "borsh", derive(BorshSerialize, BorshDeserialize))]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub enum NetworkId {
-  Serai,
+pub enum ExternalNetworkId {
   Bitcoin,
   Ethereum,
   Monero,
 }
+
+/// The type used to identify networks.
+#[derive(
+  Clone, Copy, PartialEq, Eq, Hash, Debug, Encode, Decode, PartialOrd, Ord, MaxEncodedLen, TypeInfo,
+)]
+#[cfg_attr(feature = "std", derive(Zeroize))]
+#[cfg_attr(feature = "borsh", derive(BorshSerialize, BorshDeserialize))]
```

Comment:
There's also some question of if we should similarly update Borsh. We don't need to, as Borsh is intended to be used over-the-wire as some canonical format (and is instead just used in our local DB), but we have the same commentary our wire-format flexibility.

Also, cc #573 as a relevant discussion here.

I'll also note manually defining `BorshSerailize/BorshDeserialize` to be the scale `Encode/Decode` (not replicating the same function bodies yet explicitly calling into scale) is more than fine. I'd not complain about that to de-duplicate the logic.